homeassistant/components/control4/config_flow.py

```diff
@@ -0,0 +1,205 @@
+"""Config flow for Control4 integration."""
+import logging
+
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import BadCredentials, Unauthorized
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_CONTROLLER_NAME,
+    CONF_LIGHT_COLD_START_TRANSITION_TIME,
+    CONF_LIGHT_TRANSITION_TIME,
+    DEFAULT_LIGHT_COLD_START_TRANSITION_TIME,
+    DEFAULT_LIGHT_TRANSITION_TIME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    MIN_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Control4 instances."""
+    return {
+        entry.data.get(CONF_CONTROLLER_NAME)
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class Control4Validator:
+    """Validates that config details can be used to authenticate and communicate with Control4."""
+
+    def __init__(self, host, username, password):
+        """Initialize."""
+        self.host = host
+        self.username = username
+        self.password = password
+        self.account = C4Account(self.username, self.password)
+        self.controller_name = None
+        self.director_bearer_token = None
+        self.director = None
+
+    async def authenticate(self) -> bool:
+        """Test if we can authenticate with the Control4 account API."""
+        try:
+            # Authenticate with Control4 account
+            await self.account.getAccountBearerToken()
+
+            # Get controller name
+            account_controllers = await self.account.getAccountControllers()
+            self.controller_name = account_controllers["controllerCommonName"]
+
+            # Get bearer token to communicate with controller locally
+            self.director_bearer_token = await self.account.getDirectorBearerToken(
+                self.controller_name
+            )
+            return True
+        except (BadCredentials, Unauthorized) as exception:
+            _LOGGER.error(exception)
```

Comment:
This is expected to happen.  We shouldn't log the exception especially if it contains credentials

```suggestion
        except (BadCredentials, Unauthorized):
```